src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/Program.cs

```diff
@@ -202,11 +214,28 @@ static CliRootCommand CreateCommandLineParser()
             SessionId: sessionId,
             SharedDependenciesPath: sharedDependenciesPath,
             ExtensionAssemblyPaths: extensionAssemblyPaths,
-            ExtensionLogDirectory: extensionLogDirectory);
+            ExtensionLogDirectory: extensionLogDirectory,
+            PipeName: pipe);
 
         return RunAsync(serverConfiguration, cancellationToken);
     });
 
     return rootCommand;
 }
 
+static string GetPlatformPipeConnectionString(string pipeName)
+{
+    var isWindows = RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
+    if (isWindows)
+    {
+        return @"\\.\" + pipeName;
+    }
+    else
+    {
+        // For Linux-type systems, the "pipe" is actually a shared file.
+        // Point to the same file as on the client
+        var pipeFile = pipeName + ".sock";
+        return Path.Combine(Path.GetTempPath(), pipeFile);
```

Comment:
The tempdir path should be part of the pipe name passed to the server.  In case there are any differences between node and dotnet when computing the temp path.

Reply:
Done